cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/RemoteArtifactRepositoryReader.java

```diff
@@ -153,10 +168,17 @@ public void close() throws IOException {
     List<ArtifactDetail> details = GSON.fromJson(httpResponse.getResponseBodyAsString(), ARTIFACT_DETAIL_LIST_TYPE);
     List<ArtifactDetail> detailList = new ArrayList<>();
     for (ArtifactDetail detail : details) {
-      Location artifactLocation = locationFactory.create(detail.getDescriptor().getLocationURI());
+      Location location;
+      if (artifactLocalizerClient.isPresent()) {
```

Comment:
minor nit: similarly here, if we have localizeArtifact method as  localizeArtifact(ArtifactId artifactId, Location defaultLocation), we can avoid this duplicate checks here and on line 103.

Reply:
Same as above